`__TIME__` replacement might be empty depending on compiler

The `%T` format specifier we are using to generate the timestamp is not supported by all compilers:

```
/home/runner/work/simplecpp/simplecpp/simplecpp.cpp: In function ‘std::string getTimeDefine(const tm*)’:
/home/runner/work/simplecpp/simplecpp/simplecpp.cpp:3291:34: error: ISO C++98 does not support the ‘%T’ gnu_strftime format [-Werror=format=]
 3291 |     strftime(buf, sizeof(buf), "%T", timep);
      |                                  ^
```

--- a/simplecpp.cpp
+++ b/simplecpp.cpp
@@ -3440,7 +3440,7 @@ static std::string getDateDefine(const struct tm *timep)
 static std::string getTimeDefine(const struct tm *timep)
 {
     char buf[] = "??:??:??";
-    strftime(buf, sizeof(buf), "%T", timep);
+    strftime(buf, sizeof(buf), "%H:%M:%S", timep);
     return std::string("\"").append(buf).append("\"");
 }
 

